File: sdks/typescript/src/thinking.ts
/** User-facing reasoning effort levels with Rust-compatible parsing. */
export const ThinkingLevel = Object.freeze({
  None: "none",
  Minimal: "minimal",
  Low: "low",
  Medium: "medium",
  High: "high",
  Xhigh: "xhigh",
  Max: "max",
} as const);

export type ThinkingLevel = (typeof ThinkingLevel)[keyof typeof ThinkingLevel];

export function thinkingLevelParse(value: string): ThinkingLevel | undefined {
  if (typeof value !== "string") return undefined;
  switch (value.trim().toLowerCase()) {
    case "none":
    case "off":
      return ThinkingLevel.None;
    case "minimal":
    case "min":
      return ThinkingLevel.Minimal;
    case "low":
      return ThinkingLevel.Low;
    case "medium":
    case "med":
    case "normal":
      return ThinkingLevel.Medium;
    case "high":
      return ThinkingLevel.High;
    case "xhigh":
      return ThinkingLevel.Xhigh;
    case "max":
      return ThinkingLevel.Max;
    default:
      return undefined;
  }
}

File: sdks/python/src/echo_agent_sdk/thinking.py
from __future__ import annotations

from enum import Enum


class ThinkingLevel(str, Enum):
    NONE = "none"
    MINIMAL = "minimal"
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    XHIGH = "xhigh"
    MAX = "max"

    @classmethod
    def parse(cls, value: str) -> ThinkingLevel | None:
        if not isinstance(value, str):
            return None
        return {
            "none": cls.NONE,
            "off": cls.NONE,
            "minimal": cls.MINIMAL,
            "min": cls.MINIMAL,
            "low": cls.LOW,
            "medium": cls.MEDIUM,
            "med": cls.MEDIUM,
            "normal": cls.MEDIUM,
            "high": cls.HIGH,
            "xhigh": cls.XHIGH,
            "max": cls.MAX,
        }.get(value.strip().lower())
